source/common/http/conn_pool_grid.cc

```diff
@@ -154,6 +159,15 @@ void ConnectivityGrid::WrapperCallbacks::signalFailureAndDeleteSelf(
   deleteThis();
   if (callbacks != nullptr) {
     ENVOY_LOG(trace, "Passing pool failure up to caller.");
+    std::string failure_str;
+    if (prev_tcp_pool_failure_reason_.has_value()) {
+      // TCP pool failed early on, log its error details as well.
+      failure_str = fmt::format("{} (with earlier TCP attempt failure reason {}, {})",
```

Comment:
comma might cause unexpected truncation issue. Can we use "|" instead?

Reply:
I thought you fixed our error reporting code to convert  comma to something else. This is shared Envoy code which doesn't seem to be a good place to enforce our internal policy.

Reply:
You are right. It's sanitized internally and we should be good.